### DISCUSSION TITLE
Unable to build a new package

### DISCUSSION THREAD

**Discussion by lancezhongZen** [bottlerocket-os/bottlerocket#2249] [Q&A] [answered]
Hi team, I’m trying to build a new coredns package for the Bottlerocket image following this [guide](https://github.com/bottlerocket-os/bottlerocket/tree/develop/packages/#Development), but I received the error below:

> cargo make -e BUILDSYS_ARCH=aarch64 -e PACKAGE=libcoredns build-package
...
[cargo-make] INFO - Running Task: build-package
error: package ID specification `libcoredns` did not match any packages
[cargo-make] ERROR - Error while executing command, exit code: 101
[cargo-make] WARN - Build Failed.

Here is my directory structure. 

> packages/libcoredns/
├── Cargo.toml
├── build.rs
├── pkg.rs
├── libcoredns.spec

I  made sure to refresh the Cargo.lock the file before building and execute the command in the top-level Bottlerocket directory. What is the best way to identify and resolve the issue? Thanks!

**Comment by arnaldo2792** [comment, marked as answer]
I think something missing in that guide you link is that you need to include the package in the `Cargo.toml` file of the variant you are trying to modify, so lets say you are updating `aws-k8s-1.23` you will have to modify [this file](https://github.com/bottlerocket-os/bottlerocket/blob/develop/variants/aws-k8s-1.23/Cargo.toml), and add the package to both `included-packages` and `build-dependencies`.

Once you update the `Cargo.toml` file, you need to `cd variants; cargo build; cd ../` so that `variants/Cargo.lock` is updated.

**Reply by lancezhongZen** [reply]
Tried to include the package into the Cargo.toml in one of the variants and ran `cargo generate-lockfile` in the variants directory. No more Package ID issues now. Thank you!
